helm/dagger/templates/engine-statefulset.yaml

```diff
@@ -11,14 +11,20 @@ spec:
   # DO NOT RUN MORE THAN 1 REPLICA.
   # Only one (1) Engine is able to hold the lock, other replicas will not be able to start.
   replicas: 1
+  podManagementPolicy: {{ .Values.engine.statefulSet.podManagementPolicy }}
   selector:
     matchLabels:
       name: {{ include "dagger.fullname" . }}-engine
+  {{- if and .Values.engine.storage.data.persistentVolumeClaim.enabled .Values.engine.storage.persistentVolumeClaimRetentionPolicy }}
+  persistentVolumeClaimRetentionPolicy:
+    whenDeleted: {{ .Values.engine.storage.persistentVolumeClaimRetentionPolicy.whenDeleted }}
+    whenScaled: {{ .Values.engine.storage.persistentVolumeClaimRetentionPolicy.whenScaled }}
+  {{- end }}
```

Comment:
Since these options are only used for the `statefulSet` I **think** that it would be best to namespace them, same as we did with `podManagementPolicy`. WDYT @jholm117 ?

Reply:
Definitely, good call.